src/styles/main.scss

```diff
@@ -0,0 +1,78 @@
+$gap: 1px;
+$padding: 10px;
+$day-size: 100px;
+$columns: 7;
+$days: mon, tue, wed, thu, fri, sat, sun;
+
+@use 'sass:list';
+
+.page {
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+  &__body {
+    margin: 0;
+    height: 100vh;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+  }
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: $padding;
+  width: calc((#{$day-size} * #{$columns}) + (#{$gap} * #{$columns - 1}));
+  background-color: #fff;
+
+  &__day {
+    width: $day-size;
+    height: $day-size;
+    background-color: #eee;
+    display: flex;
+    border: 1px solid #000;
+    box-sizing: border-box;
+    transition:
+      background-color 0.5s,
+      transform 0.5s;
+
+    &::before {
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      margin: auto;
+      font-family: Arial, sans-serif;
+      font-size: 30px;
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transform: translateY(-20px);
+      transition:
+        background-color 0.5s,
+        transform 0.5s;
```

Comment:
You have already specified transition in the parent selector so you don't need it here